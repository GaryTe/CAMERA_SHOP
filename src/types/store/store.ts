import { store } from '../../store/store/store';
import { listCamera, CardProduct, Reviews, PromoProduct, Review } from '../types-response/types-response';

export type Cameras = {
  cameras: listCamera | null;
  modifiedListProduct: listCamera | null;
}

export type CardCamera = {
  dataCamera: CardProduct | null;
}

export type SimilaCameras = {
  dataSimilaCameras: listCamera | null;
  sortListCamerra: listCamera | null;
  counterMin: number;
  counterMax: number;
}

export type ReviewsCamera = {
  dataReviews: Reviews | Array<Review>;
  resultSortReviews: Reviews | null;
  reviews: Reviews | null;
  counter: number;
}

export type PromoCamera = {
  dataPromoCamera: PromoProduct | null;
}

export type Cart = {
  dataProduct: listCamera | Array<CardProduct>;
  loading: boolean;
}

export type RootState = ReturnType<typeof store.getState>;
