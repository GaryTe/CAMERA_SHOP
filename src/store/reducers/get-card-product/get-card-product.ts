import { createSlice } from '@reduxjs/toolkit';
import { fetchDataCardProduct } from '../../api-actions/api-actions';
import { CardCamera } from '../../../types/store/store';
import { NameSpace } from '../../../const/const';
import { product } from '../../../util/mocks';

const initialState: CardCamera = {
  dataCamera: null
};

export const getCardProduct = createSlice({
  name: NameSpace.CardProduct,
  initialState,
  reducers: {
    functionForTestingReducers: (state) => {
      state.dataCamera = product;
    }
  },
  extraReducers(builder) {
    builder
      .addCase(fetchDataCardProduct.fulfilled, (state, action) => {
        state.dataCamera = action.payload;
      });
  }
});

export const {functionForTestingReducers} = getCardProduct.actions;
