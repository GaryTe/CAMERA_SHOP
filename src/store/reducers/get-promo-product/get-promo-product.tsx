import { createSlice } from '@reduxjs/toolkit';
import { fetchDataPromoProduct } from '../../api-actions/api-actions';
import { PromoCamera } from '../../../types/store/store';
import { NameSpace } from '../../../const/const';
import { promoProduct } from '../../../util/mocks';

const initialState: PromoCamera = {
  dataPromoCamera: null
};

export const getPromoProduct = createSlice({
  name: NameSpace.Promo,
  initialState,
  reducers: {
    functionForTesting: (state) => {
      state.dataPromoCamera = promoProduct;
    }
  },
  extraReducers(builder) {
    builder
      .addCase(fetchDataPromoProduct.fulfilled, (state, action) => {
        state.dataPromoCamera = action.payload;
      });
  }
});

export const {functionForTesting} = getPromoProduct.actions;
