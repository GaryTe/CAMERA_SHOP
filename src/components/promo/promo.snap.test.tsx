import {render} from '@testing-library/react';
import {Provider} from 'react-redux';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import Promo from './promo';
import { functionForTesting } from '../../store/reducers/get-promo-product/get-promo-product';
import { functionForTestingReducers } from '../../store/reducers/get-card-product/get-card-product';
import { store } from '../../store/store/store';

const history = createMemoryHistory();

const children = (
  <Promo
    requestDataPromoProduct={functionForTesting}
    requestDataCardProduct={functionForTestingReducers}
  />
);

describe('Component: Promo', () => {
  it('should render Promo', () => {

    const {container} = render(
      <Provider store={store}>
        <Router
          location={history.location}
          navigationType={history.action}
          navigator={history}
        >
          {children}
        </Router>
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
