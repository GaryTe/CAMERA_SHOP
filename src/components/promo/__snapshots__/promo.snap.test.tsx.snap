// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Promo should render Promo 1`] = `
<div>
  <div
    class="banner"
  >
    <picture>
      <source
        src="img/content/promo.webp"
        srcset="img/content/promo@2x.webp"
        type="image/webp"
      />
      <img
        alt="баннер"
        height="280"
        src="img/content/promo.jpg"
        srcset="img/content/promo@2x.jpg"
        width="1280"
      />
    </picture>
    <p
      class="banner__info"
    >
      <span
        class="banner__message"
      >
        Новинка!
      </span>
      <span
        class="title title--h1"
      >
        Ретрокамера Dus Auge lV
      </span>
      <span
        class="banner__text"
      >
        Нулевой
         
        Видеокамера
         от известного производителя
      </span>
      <a
        class="btn"
        href="/product/1"
      >
        Подробнее
      </a>
    </p>
  </div>
</div>
`;
